src/Raven.Server/Documents/Handlers/AttachmentHandler.cs

```diff
@@ -387,6 +387,182 @@ public async Task Delete()
             }
         }
 
+        [RavenAction("/databases/*/debug/attachments/missing", "GET", AuthorizationStatus.ValidUser, EndpointType.Read, DisableOnCpuCreditsExhaustion = true)]
+        public async Task GetMissingAttachments()
+        {
+            var types = new List<AttachmentType>(2);
+            var typeString = GetStringQueryString("type", required: false);
+            var collection = GetStringQueryString("collection", required: true);
+            if (string.IsNullOrEmpty(typeString) == false)
+            {
+                if (Enum.TryParse(typeString, true, out AttachmentType type) == false)
+                {
+                    throw new ArgumentException($"Query string '{typeString}' was not recognized as valid type");
+                }
+
+                types.Add(type);
+            }
+            else
+            {
+                types.Add(AttachmentType.Document);
+                types.Add(AttachmentType.Revision);
+            }
+
+            if (string.IsNullOrEmpty(collection) == false)
+            {
+                if (collection.Equals(Constants.Documents.Collections.AllDocumentsCollection, StringComparison.OrdinalIgnoreCase) == false)
+                {
+                    if (Database.DocumentsStorage.GetCollection(collection, true) == null)
+                        throw new ArgumentException($"Query string '{collection}' was not recognized as valid collection name");
+                }
+            }
+
+            var start = GetLongQueryString("start", required: false) ?? 0;
+            var pageSize = GetPageSize();
+
+            using (var token = CreateHttpRequestBoundOperationToken())
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            using (context.OpenReadTransaction())
+            await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                writer.WriteStartObject();
+
+                if (types.Contains(AttachmentType.Revision))
+                {
+                    writer.WritePropertyName("Revisions");
+                    writer.WriteStartObject();
+                    WriteMissingAttachments(Database, Database.DocumentsStorage.RevisionsStorage.GetRevisionsFrom(context, start, pageSize), context,  writer, AttachmentType.Revision, token);
```

Comment:
This can be _super_ expensive in terms of I/O & time, no?
That being the case - we may need to output some values to the response to avoid timeouts?